src/Bicep.Core/Registry/AzureContainerRegistryManager.cs

```diff
@@ -235,13 +248,14 @@ private static async Task<Stream> ProcessLayer(ContainerRegistryContentClient cl
 
         private static void ProcessConfig(OciDescriptor config)
         {
+            string[] allowedMediaTypes = { BicepMediaTypes.BicepModuleConfigV1, BicepMediaTypes.BicepProviderConfigV1 };
             // media types are case insensitive
-            if (!string.Equals(config.MediaType, BicepMediaTypes.BicepModuleConfigV1, MediaTypeComparison))
+            if (!allowedMediaTypes.Contains(config.MediaType, StringComparer.OrdinalIgnoreCase))
             {
                 throw new InvalidModuleException($"Did not expect config media type \"{config.MediaType}\".");
             }
-
-            if (config.Size != 0)
+            
+            if (config.Size > 2)
```

Comment:
>2 [](http://example.com/codeflow?start=30&length=1)

What is the significance of 2 here?

Reply:
ORAS will publish config as `{}` (an empty json object) with two chars so the size is 2

Reply:
We don't have a user-facing use case for consuming artifacts pushed by ORAS. If it's helpful for development, you could probably give `NUL` file to ORAS (if on Windows) to produce a null config blob when pushing.

That said I don't have a huge objection to allow {}, but checking if size is greater than 2 is really fragile. If we're allowing "{}" in the config blob, why not " { } "? For simplicity's sake, should we just remove the config blob check entirely?

If you want to keep the check, then we should probably allow either null or {} blob but we should deserialize it into JSON and then check that it's an empty object.